src/test/java/org/apache/maven/plugins/stage/RepositoryCopierTest.java

```diff
@@ -19,46 +19,150 @@
+    private void startFtpServer()
+    {
+        sshd = SshServer.setUpDefaultServer();
+        sshd.setPort( PORT );
+
+        Security.addProvider( new BouncyCastleProvider() );
+        AbstractFileKeyPairProvider fileKeyPairProvider = SecurityUtils.createFileKeyPairProvider();
+        fileKeyPairProvider.setFiles(
+            Collections.singletonList( new File( getBasedir(), KEY) )
+        );
+        sshd.setKeyPairProvider( fileKeyPairProvider );
+
+        sshd.setFileSystemFactory(new VirtualFileSystemFactory() {
+            @Override
+            public Path getDefaultHomeDir()
+            {
+                return serverFileSystemRoot();
+            }
+        });
+
+        List<NamedFactory<UserAuth>> userAuthFactories = new ArrayList<>();
+        userAuthFactories.add( new UserAuthNoneFactory() );
+        sshd.setUserAuthFactories( userAuthFactories );
+        sshd.setPublickeyAuthenticator( AcceptAllPublickeyAuthenticator.INSTANCE );
+
+        final ScpCommandFactory scpCommandFactory = new ScpCommandFactory();
+        scpCommandFactory.setDelegateCommandFactory( new FakeUnixCommandFactory( serverFileSystemRoot() ) );
+        sshd.setCommandFactory(scpCommandFactory);
+
+        List<NamedFactory<Command>> namedFactoryList = new ArrayList<>();
+        namedFactoryList.add( new SftpSubsystemFactory() );
+        sshd.setSubsystemFactories( namedFactoryList );
+        try
+        {
+            sshd.start();
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
 
-        File targetRepo = new File( getBasedir(), "target/target-repository" );
+    // To make clean-up easier, copy the entire target repository to a new working copy. That way, rather than worry about what
+    // modifications we might have made, we can simply delete the whole thing when we're done.
+    private void makeWorkingRepository()
+    {
+        final File prototype = new File( getBasedir(), TARGET_REPOSITORY );
+        final File workingDir = new File( getBasedir(), WORKING_TARGET_REPOSITORY );
+        try
+        {
+            FileUtils.copyDirectory( prototype, workingDir );
+        }
+        catch ( IOException e )
+        {
+            throw new RuntimeException( "Couldn't copy target repository directory", e );
+        }
+    }
 
-        System.out.println( "Copying target stage for tests ..." );
+    private void deleteWorkingRepository() throws IOException
+    {
+        FileUtils.deleteDirectory( new File( getBasedir(), WORKING_TARGET_REPOSITORY ) );
+    }
 
-        FileUtils.copyDirectoryStructure( targetRepoSource, targetRepo );
+    private Path serverFileSystemRoot()
+    {
+        return new File( getBasedir(), WORKING_TARGET_REPOSITORY ).toPath();
+    }
 
-        File stagingRepo = new File( getBasedir(), "src/test/staging-repository" );
+    public void tearDown() throws Exception
+    {
+        super.tearDown();
```

Comment:
super.tearDown should come last in the method

Reply:
Thanks, done.